packages/office-ui-fabric-react/src/components/Rating/Rating.tsx

```diff
@@ -120,8 +122,21 @@ export class Rating extends BaseComponent<IRatingProps, IRatingState> {
     );
   }
 
-  private _getIcon(): JSX.Element {
-    return <i className={ css('ms-Icon', this.props.icon || 'ms-Icon--FavoriteStarFill') } />;
+  private _getIcon(starSize?: number): JSX.Element {
+    if (starSize && starSize < 1) {
+      return (
+        <span className={ css('ms-rating-halfStarContainer', styles.halfStarContainer, 'ms-Icon', this.props.icon || 'ms-Icon--FavoriteStarFill') } >
+          <i className={ css('ms-Rating-star', styles.star,
+            'is-selected ' + styles.starIsSelected,
+            { ['is-disabled ' + styles.starIsDisabled]: this.props.disabled },
+            'ms-rating-halfStar', styles.halfStar, 'ms-Icon',
+            this.props.icon || 'ms-Icon--FavoriteStarFill') }
+            style={ { width: starSize * 100 + '%' } } />
```

Comment:
If I'm understanding this right, it's not always a half star but instead we hide some portion of the star? So 10% or 90% of the star might be visible? This could be hard to make sense of, and I think we'd most likely want to force it to round to half/50% of a star.

There is a half star icon that we could have added to the icon set, which would greatly simplify all of this. Until then, I'd recommend that we always round to the nearest half star. So 1.9 = 2, 1.6 = 1.5, etc.